src/components/Time/MainTimeControl.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import TimeControlsContainer from './TimeControls';
+import TimeRangeComponent from './TimeRangeComponent';
+import RefreshContainer from '../Refresh/Refresh';
+
+type Props = {
+  customDuration: boolean;
+};
+
+export default class MainTimeControl extends React.Component<Props> {
```

Comment:
Naming is always hard but I do find it a little confusing to have MainTimeControl and also TimeControls.   Maybe this could just be "TimeControl" if we were to rename TimeControls to be more in line with TimeRangeComponent.  TimeRangeComponent is well named, it indicates that you can set a complete time range (aka custom time range).  TimeControls is the component that is a little weird, maybe it should be DurationComponent to make the difference more explicit.  I realize that it also includes the Refresh whereas TimeRangeComponent does not.  Anyway, just an observation, not a blocker.

Reply:
+1
TimeControl, and TimeDuration(Component) and TimeRange(Component) sounds like a good way.